### ISSUE TITLE
Data export issue

### ISSUE THREAD

**Issue by suchow** [Dallinger/Dallinger#640]
# 1

I tested PR #630 by:

1. Running Griduniverse with 1 participant: 

```
dallinger debug --verbose
```
2. Exporting the dataset:

```
dallinger export --local --app 9b933a1e-0091-4b47-aee8-3b43cbfa4f69
```

3. Loading the dataset:

```
dallinger load data/9b933a1e-0091-4b47-aee8-3b43cbfa4f69-data.zip
```

I get the following error:

```
psycopg2.IntegrityError: insert or update on table "node" violates foreign key constraint "node_participant_id_fkey"
DETAIL:  Key (participant_id)=(1) is not present in table "participant".
```

Looking into this, I see that the exported dataset has a zero-byte participant table, which is unexpected and does not match the data at the moment before the export. I can see that the Participant object is correctly created during the experiment.

This may have something to do with a failure to save the table during the process that scrubs it of PII.

**Comment by suchow** [comment]
@jessesnyder says:

> @suchow I did encounter #1 while working on the story. I was able to get a successful export by importing and running the export() function while the experiment was still running. At the time all I cared about was getting a good export, and thought I might have been doing something wrong. If not, I'd say this is a bug in the export feature.

**Comment by alecpm** [comment]
Looks like there was a race condition in the closing of files when scrubbing participants. Should be fixed as part of pull request #630 (tests for fix dependent on data import into DB).

**Comment by suchow** [comment]
Closed by #630.
